rust/src/storage/s3/dynamodb_lock.rs

```diff
@@ -400,7 +423,7 @@ impl DynamoDbLockClient {
         Ok(LockItem {
             owner_name: self.opts.owner_name.clone(),
             record_version_number: rvn,
-            lease_duration: self.opts.lease_duration,
+            lease_duration: Some(self.opts.lease_duration),
```

Comment:
so upsert a nonexpirable lock will make it expirable? what's the use-case for this?

Reply:
There's no use case for non-expirable lock in delta-rs. I was thinking of adding that, e.g `acquire_non_expirable_lock` but since we don't have a use case for that then left it as is

Reply:
got it, I am mostly concerned about a future client accidentally invoked one of the upsert method to corrupt the lock.

how does KDI acquire a nonexpirable lock to block writes from spark in this case?

Reply:
> concerned about a future client accidentally invoked one of the upsert method to corrupt the lock.

It's private so should not be possible. And other functions that calls it will succeed unless the lock is owned.

> how does KDI acquire a nonexpirable lock to block writes from spark in this case?

It's not, we plan to do that within proprietary code

Reply:
this looks good based on our offline discussion, looks like it is not going cause problem because optimize will just ignore the lease attribute entirely and never acquire an expired lock from its end.